relay-server/src/metrics.rs

```diff
@@ -0,0 +1,200 @@
+use relay_common::metrics::{CounterMetric, HistogramMetric, SetMetric, TimerMetric};
+
+/// Set metrics used by Relay
+pub enum RelaySets {
+    /// Represents the number of active projects in the current slice of time
+    UniqueProjects,
+}
+
+impl SetMetric for RelaySets {
+    fn name(&self) -> &'static str {
+        match self {
+            RelaySets::UniqueProjects => "unique_projects",
+        }
+    }
+}
+
+/// Histogram metrics used by Relay.
+pub enum RelayHistograms {
+    /// The number of events in the queue as a percentage of the maximum number of events
+    /// that can be stored in the queue ( 0 ... the queue is empty, 1 ... the queue is full
+    /// and no additional events can be added).
+    EventQueueSizePct,
+    /// The number of events in the queue at the sampling moment.
+    EventQueueSize,
+    /// The event size as seen by Relay after it is extracted from a request
+    EventSizeBytesRaw,
+    /// The event size as seen by Relay after it is extracted Base64 decoded and unzipped.
+    EventSizeBytesUncompressed,
+    /// Number of projects in the ProjectCache that are waiting for their state to be updated.
+    ProjectStatePending,
+    /// Number of project state requested from the Upstream for the current batch request.
+    ProjectStateRequestBatchSize,
+    /// Number of project states received from the Upstream for the current batch request.
+    ProjectStateReceived,
+    /// Number of project states currently held in the ProjectState cache.
+    ProjectStateCacheSize,
+}
+
+impl HistogramMetric for RelayHistograms {
+    fn name(&self) -> &'static str {
+        match self {
+            RelayHistograms::EventQueueSizePct => "event.queue_size.pct",
+            RelayHistograms::EventQueueSize => "event.queue_size",
+            RelayHistograms::EventSizeBytesRaw => "event.size_bytes.raw",
+            RelayHistograms::EventSizeBytesUncompressed => "event.size_bytes.uncompressed",
+            RelayHistograms::ProjectStatePending => "project_state.pending",
+            RelayHistograms::ProjectStateRequestBatchSize => "project_state.request.batch_size",
+            RelayHistograms::ProjectStateReceived => "project_state.received",
+            RelayHistograms::ProjectStateCacheSize => "project_cache.size",
+        }
+    }
+}
+
+/// Timer metrics used by Relay
+pub enum RelayTimers {
+    /// The time spent deserializing an event from a JSON byte array into a `relay_general::protocol::Event`.
```

Comment:
Since this is external documentation, let's not refer to code types too much. 
```suggestion
    /// The time spent deserializing an event from JSON into memory.
```